src/components/forms/ClusteredDiskSizeSelector.tsx

```diff
@@ -0,0 +1,231 @@
+                            : ""
+                        }
+                        className="u-no-margin--bottom"
+                      />
+                      <Select
+                        id={
+                          memberNames.indexOf(item) === 0
+                            ? "memUnitSelect"
+                            : `memUnitSelect-${item}`
+                        }
+                        name="memUnitSelect"
+                        label="Select disk size unit"
+                        labelClassName="u-off-screen"
+                        options={getMemUnitOptions()}
+                        onChange={(e) =>
+                          setValueForMember(
+                            `${limit(activeValue || "").value ?? 0}${e.target.value}`,
+                            item,
+                          )
+                        }
+                        value={limit(activeValue || "").unit}
+                        disabled={disabled}
+                        className="u-no-margin--bottom"
+                      />
+                    </div>
+                  )}
+                </div>
+              </Fragment>
+            );
+          })}
+          {helpText && (
+            <div className="p-form-help-text cluster-specific-helptext">
+              {helpText}
+            </div>
+          )}
+        </div>
+      )}
+
+      {!isSpecific && (
+        <div>
+          {isReadOnly ? (
+            <>
+              {firstValue}
+              <FormEditButton toggleReadOnly={toggleReadOnly} />
+            </>
+          ) : (
+            <>
+              <div className="memory-limit-with-unit">
+                <Input
+                  id={id}
+                  name="limits_disk"
+                  type="number"
+                  min="0"
+                  step="Any"
+                  placeholder="Enter value"
+                  onChange={(e) =>
+                    setValueForAllMembers(
+                      e.target.value + limit(firstValue || "").unit,
+                    )
+                  }
+                  value={
+                    firstValue?.match(/^\d/)
+                      ? limit(firstValue || "").value
+                      : ""
+                  }
+                  disabled={disabled}
+                />
+                <Select
+                  id="memUnitSelect"
+                  name="memUnitSelect"
+                  label="Select disk size unit"
+                  labelClassName="u-off-screen"
+                  options={getMemUnitOptions()}
+                  onChange={(e) =>
+                    setValueForAllMembers(
+                      `${limit(firstValue || "").value ?? 0}${e.target.value}`,
+                    )
+                  }
+                  value={limit(firstValue || "").unit}
+                  disabled={disabled}
+                />
+              </div>
+              {helpText && <div className="p-form-help-text">{helpText}</div>}
```

Comment:
I think instead of repeating this in here twice, we should invoke the `DiskSizeSelector` component twice. That should make this component way slimmer, as we can also remove some of the helper functions.